### DISCUSSION TITLE
How to change scenecut-aware-qpconformance-window-offsets in x265 in 2.4.0

### DISCUSSION THREAD

**Discussion by ACKERINIHO** [staxrip/staxrip#683] [Q&A] [answered]
Earlier i had version 2.1.7.0 and that had this set to no-scenecut-aware-qpconformance-window-offsets but now in 2.4.0 its set to  scenecut-aware-qpconformance-window-offsets and i can't find any option in settings that changes it back to original.

**Comment by Dendraspis** [comment, marked as answer]
It's in **Rate Control 3**, last option.

Btw:  Those Settings are disabled by default, so it seems you use old settings, which is not recommended, or you forgot you set them.

**Reply by ACKERINIHO** [reply]
that is unticked for me still its activated.

**Reply by Dendraspis** [reply]
Do it seems you use your settings file from an old version (probably v2.1.7.0). You should delete it and start with new settings.

**Reply by ACKERINIHO** [reply]
Hello i did what you said started with complete new settings and made sure that option was unticked i am still getting scenecut-aware-qpconformance-window-offsets and not no-scenecut-aware-qpconformance-window-offsets . do you think this might be a bug?

**Reply by Dendraspis** [reply]
Can you verify your statements with screenshots? Where do you see those params? What have you changed, because a default x265 setting has them disabled.

And no, it's probably not a bug.

**Reply by ghost** [reply]
If your settings are new, but still using an old template or an old project, then you could have that kind of strange things.

**Reply by ACKERINIHO** [reply]
> Can you verify your statements with screenshots? Where do you see those params? What have you changed, because a default x265 setting has them disabled.
> 
> And no, it's probably not a bug.

its only visible in mediainfo after the fact and not visible during the encoding process. ill try to post screenshot later. to replicate this encode any length file with x265 its usually the 5th last parameter

 

> If your settings are new, but still using an old template or an old project, then you could have that kind of strange things.

my template is also completely new

**Reply by ACKERINIHO** [reply]
![stax](https://user-images.githubusercontent.com/83504278/116800860-5ea45d80-ab22-11eb-80a1-23d70a5554bd.png)
this pic is taken with a fresh download of 2.4.0 with nothing change just x265 set to slow. no templates or settings changes

**Reply by ghost** [reply]
Ah now it is clear what you mean ! I thought there was a problem in the COMMAND LINE. So it's only in the media info TAG.
This could be a simple tagging problem. Investigating...

**Reply by ghost** [reply]
OK this tag is present in both latest builds of @Patman86 and @DJATOM.

Here's the command:
`"C:\Program Files\StaxRip\Apps\Encoders\x265\x265.exe" --reader-options library="C:\Program Files\StaxRip\Apps\FrameServer\AviSynth\AviSynth.dll" --crf 18 --output-depth 10 --colorprim bt709 --colormatrix bt709 --transfer bt709 --range limited --sar 1:1 --output C:\Temp\_StaxRip\TimecodeSample25Fps_Chapts_temp\TimecodeSample25Fps_Chapts_out.hevc C:\Temp\_StaxRip\TimecodeSample25Fps_Chapts_temp\TimecodeSample25Fps_Chapts.avs`


Here is the media info relevant part ("Encoding Settings"), look at last line or line before last:

`cpuid=1111039 / frame-threads=3 / numa-pools=8 / wpp / no-pmode / no-pme /  no-psnr / no-ssim / log-level=2 / input-csp=1 / input-res=1280x720 / interlace=0  / total-frames=1960 / level-idc=0 / high-tier=1 / uhd-bd=0 / ref=3 /  no-allow-non-conformance / no-repeat-headers / annexb / no-aud / no-hrd / info /  hash=0 / no-temporal-layers / open-gop / min-keyint=25 / keyint=250 /  gop-lookahead=0 / bframes=4 / b-adapt=2 / b-pyramid / bframe-bias=0 /  rc-lookahead=20 / lookahead-slices=4 / scenecut=40 / hist-scenecut=0 / radl=0 /  no-splice / no-intra-refresh / ctu=64 / min-cu-size=8 / no-rect / no-amp /  max-tu-size=32 / tu-inter-depth=1 / tu-intra-depth=1 / limit-tu=0 / rdoq-level=0  / dynamic-rd=0.00 / no-ssim-rd / signhide / no-tskip / nr-intra=0 / nr-inter=0 /  no-constrained-intra / strong-intra-smoothing / max-merge=3 / limit-refs=3 /  no-limit-modes / me=1 / subme=2 / merange=57 / temporal-mvp / no-frame-dup /  no-hme / weightp / no-weightb / no-analyze-src-pics / deblock=0:0 / sao /  no-sao-non-deblock / rd=3 / selective-sao=4 / early-skip / rskip / no-fast-intra  / no-tskip-fast / no-cu-lossless / b-intra / no-splitrd-skip / rdpenalty=0 /  psy-rd=2.00 / psy-rdoq=0.00 / no-rd-refine / no-lossless / cbqpoffs=0 /  crqpoffs=0 / rc=crf / crf=18.0 / qcomp=0.60 / qpstep=4 / stats-write=0 /  stats-read=0 / ipratio=1.40 / pbratio=1.30 / aq-mode=2 / aq-strength=1.00 /  cutree / zone-count=0 / no-strict-cbr / qg-size=32 / no-rc-grain / qpmax=69 /  qpmin=0 / no-const-vbv / sar=1 / overscan=0 / videoformat=5 / range=0 /  colorprim=1 / transfer=1 / colormatrix=1 / chromaloc=0 / display-window=0 /  cll=0,0 / min-luma=0 / max-luma=1023 / log2-max-poc-lsb=8 / vui-timing-info /  vui-hrd-info / slices=1 / no-opt-qp-pps / no-opt-ref-list-length-pps /  no-multi-pass-opt-rps / scenecut-bias=0.05 / hist-threshold=0.03 /  no-opt-cu-delta-qp / no-aq-motion / no-hdr10 / no-hdr10-opt / no-dhdr10-opt /  no-idr-recovery-sei / analysis-reuse-level=0 / analysis-save-reuse-level=0 /  analysis-load-reuse-level=0 / scale-factor=0 / refine-intra=0 / refine-inter=0 /  refine-mv=1 / refine-ctu-distortion=0 / no-limit-sao / ctu-info=0 /  no-lowpass-dct / refine-analysis-type=0 / copy-pic=1 / max-ausize-factor=1.0 /  no-dynamic-refine / no-single-sei / no-hevc-aq / no-svt / no-field /  qp-adaptation-range=1.00 / scenecut-aware-qp=0conformance-window-offsets /  right=0 / bottom=0 / decoder-max-rate=0 / no-vbv-live-multi-pass`



What does `scenecut-aware-qp=0conformance-window-offsets` mean? it seems like nonesense, I hope I'm wrong.

**Reply by ghost** [reply]
@ACKERINIHO this is related to a TOOL, not an issue with staxrip. It appears to be a problem with tagging, so your encodings are probably fine. You can follow it here:
https://github.com/DJATOM/x265-aMod/issues/14

**Reply by ACKERINIHO** [reply]
so you mean the setting itself is not on but it is being tagged as so in the mediainfo? thanks for raising the issue.

**Reply by Patman86** [reply]
This failure only exists in DJATOMs and my mod?
The definitions are in the x265 source: [param.cpp line 2339 - 2342](https://bitbucket.org/multicoreware/x265_git/src/bf91444e034831141e0ce02b1200e51996f8b6c6/source/common/param.cpp#lines-2339)

**Reply by ACKERINIHO** [reply]
well in 2.1.7.0 i ported the the x265 from 2.0.8.0 so i am sure whether it was introduced after 2.1.7.0 or after 2.0.8.0 but i am sure the tag should be no-scenecut-aware-qpconformance-window-offsets. either the checkbox isn't doing anything or its mistagged.

**Reply by Dendraspis** [reply]
@Patman86 
No, not "only".  😉   @44vince44  just said, that this behavior is present in both mods, without judging about the source of the issue.
So it seems to be just a missing *whitespace* in [line 2333](https://github.com/Patman86/x265-Mod-by-Patman/blob/master/source/common/param.cpp#L2333).


> but i am sure the tag should be no-scenecut-aware-qpconformance-window-offsets. either the checkbox isn't doing anything or its mistagged.

No, when turned off, it should just say ` scenecut-aware-qp=0 conformance-window-offsets right=...`.

**Reply by Patman86** [reply]
I saw it that way too. Have already changed it locally, will test it.

**Reply by ghost** [reply]
@Patman86 it is in both your build and DJATOM's

I added a comment to the thread at DJATOM's page

**Reply by ACKERINIHO** [reply]
> No, when turned off, it should just say ` scenecut-aware-qp=0 conformance-window-offsets right=...`.

strange then why was it no-scenecut-aware-qpconformance-window-offsets before i figured the no- part specified when both these tags were off.

**Reply by Dendraspis** [reply]
@44vince44  
Yes, but the origin of this issue is the original x265 source code. Just to clarify.

@Patman86 
Do you plan to release a new version (with other updates/fixes) these days?

@ACKERINIHO 
Don't forget, that recently there was an update from v3.4 to 3.5 - so maybe they changed the output there.

**Reply by Patman86** [reply]
I want to update some tools today!

**Reply by Dendraspis** [reply]
@Patman86 
Perfect timing.  😁

**Reply by ghost** [reply]
@ACKERINIHO:

Exactly as @Dendraspis explained above, when versions of x265 evolve, some switches evolve, and some tags evolve too.
And this is NOT STAXRIP business, not even @Patman86 and @DJATOM. It is from the original x265 project.
However, you allowed us to spot a "formatting" issue, and thank you for that.

> i figured the no- part specified when both these tags were off.

Now that the formatting of the tag will (hopefully) be fixed, there won't be any confusion.

**Reply by ACKERINIHO** [reply]
ah ok i suppose it won't affect the encoding. thanks for your help

**Reply by Patman86** [reply]
I made the changes in my mod, mediainfo now shows the following:
`Encoding settings           : cpuid=1111039 / frame-threads=4 / numa-pools=16 / wpp / no-pmode / no-pme / no-psnr / no-ssim / log-level=2 / input-csp=1 / input-res=720x304 / interlace=0 / total-frames=4953 / level-idc=0 / high-tier=1 / uhd-bd=0 / ref=3 / no-allow-non-conformance / no-repeat-headers / annexb / no-aud / no-eob / no-eos / no-hrd / info / hash=0 / no-temporal-layers / open-gop / min-keyint=24 / keyint=250 / gop-lookahead=0 / bframes=4 / b-adapt=2 / b-pyramid / bframe-bias=0 / rc-lookahead=20 / lookahead-slices=0 / scenecut=40 / hist-scenecut=0 / radl=0 / no-splice / no-intra-refresh / ctu=64 / min-cu-size=8 / no-rect / no-amp / max-tu-size=32 / tu-inter-depth=1 / tu-intra-depth=1 / limit-tu=0 / rdoq-level=0 / dynamic-rd=0.00 / no-ssim-rd / signhide / no-tskip / nr-intra=0 / nr-inter=0 / no-constrained-intra / strong-intra-smoothing / max-merge=3 / limit-refs=3 / no-limit-modes / me=1 / subme=2 / merange=57 / temporal-mvp / no-frame-dup / no-hme / weightp / no-weightb / no-analyze-src-pics / deblock=0:0 / sao / no-sao-non-deblock / rd=3 / selective-sao=4 / early-skip / rskip / no-fast-intra / no-tskip-fast / no-cu-lossless / b-intra / no-splitrd-skip / rdpenalty=0 / psy-rd=2.00 / psy-rdoq=0.00 / no-rd-refine / no-lossless / cbqpoffs=0 / crqpoffs=0 / rc=crf / crf=18.0 / qcomp=0.60 / qpstep=4 / stats-write=0 / stats-read=0 / ipratio=1.40 / pbratio=1.30 / aq-mode=2 / aq-strength=1.00 / cutree / zone-count=0 / no-strict-cbr / qg-size=32 / no-rc-grain / qpmax=69 / qpmin=0 / no-const-vbv / sar=0 / overscan=0 / videoformat=5 / range=0 / colorprim=9 / transfer=16 / colormatrix=9 / chromaloc=0 / display-window=0 / cll=0,0 / min-luma=0 / max-luma=1023 / log2-max-poc-lsb=8 / vui-timing-info / vui-hrd-info / slices=1 / no-opt-qp-pps / no-opt-ref-list-length-pps / no-multi-pass-opt-rps / scenecut-bias=0.05 / hist-threshold=0.03 / no-opt-cu-delta-qp / no-aq-motion / no-hdr10 / no-hdr10-opt / no-dhdr10-opt / no-idr-recovery-sei / analysis-reuse-level=0 / analysis-save-reuse-level=0 / analysis-load-reuse-level=0 / scale-factor=0 / refine-intra=0 / refine-inter=0 / refine-mv=1 / refine-ctu-distortion=0 / no-limit-sao / ctu-info=0 / no-lowpass-dct / refine-analysis-type=0 / copy-pic=1 / max-ausize-factor=1.0 / no-dynamic-refine / no-single-sei / no-hevc-aq / no-svt / no-field / qp-adaptation-range=1.00 / scenecut-aware-qp=0 / conformance-window-right-offset=0 / conformance-window-bottom-offset=0 / decoder-max-rate=0 / no-vbv-live-multi-pass`

**Reply by ghost** [reply]
Thanks @Patman86 !!!

**Reply by Patman86** [reply]
Tools updated...

**Reply by fabricionaweb** [reply]
First of all, sorry to bump this old topic. But I thought would be better to follow the context.

> I made the changes in my mod, mediainfo now shows the following:

Im using default presets, I guess its "--no-scenecut-aware-qp" and building the latest version `x265-4.1+212+35-87f5a69a8` (on MacOS if that matters, I dont think so) and I still see it wrong: `scenecut-aware-qp=0conformance-window-offsets`

shouldn't have an leading space here? [source/common/param.cpp#L2498](https://github.com/Patman86/x265-Mod-by-Patman/blob/master/source/common/param.cpp#L2498)

**Reply by Dendraspis** [reply]
It would have been better to open an issue in Patman86's repo, but I have forwarded it to him.  Fix should be easy.  😅 

> shouldn't have an leading space here?

You are absolutely right.

**Reply by Patman86** [reply]
And this space is also missing in Main! Fortunately, it's not my mistake 😂 I've corrected it in my repo.
